repo/config/datastore.go

```diff
@@ -1,40 +1,43 @@
 package config
 
-import (
-	"encoding/json"
-)
-
 // DefaultDataStoreDirectory is the directory to store all the local IPFS data.
 const DefaultDataStoreDirectory = "datastore"
 
 // Datastore tracks the configuration of the datastore.
 type Datastore struct {
-	Type               string
-	Path               string
 	StorageMax         string // in B, kB, kiB, MB, ...
 	StorageGCWatermark int64  // in percentage to multiply on StorageMax
 	GCPeriod           string // in ns, us, ms, s, m, h
+	Path               string
+	NoSync             bool // deprecated
+
+	Spec map[string]interface{}
 
-	Params          *json.RawMessage
-	NoSync          bool
 	HashOnRead      bool
 	BloomFilterSize int
 }
 
-func (d *Datastore) ParamData() []byte {
-	if d.Params == nil {
-		return nil
-	}
-
-	return []byte(*d.Params)
-}
-
 type S3Datastore struct {
 	Region string `json:"region"`
 	Bucket string `json:"bucket"`
 	ACL    string `json:"acl"`
 }
 
+type FlatDS struct {
+	Path      string
+	ShardFunc string
+	Sync      bool
```

Comment:
So we changed this from NoSync to Sync, was this intentional?  Is the default now not to enable Sync?